browser/src/core/Socket.js

```diff
@@ -493,7 +493,8 @@ app.definitions.Socket = L.Class.extend({
 	// buffer of web-socket messages in the client that we can't
 	// process so - slurp and then emit at idle - its faster to delay!
 	_slurpMessage: function(e) {
-		this._extractTextImg(e);
+		if (!this._extractTextImg(e))
```

Comment:
Would be good to add comment above `_extractTextImg` definition to say what the boolean it returns means.